jenkins/spark-premerge-build.sh

```diff
@@ -154,7 +154,18 @@ ci_2() {
 
     # Download a Scala 2.12 build of spark
     prepare_spark $SPARK_VER 2.12
-    ./integration_tests/run_pyspark_from_build.sh
+
+
+    # Temporarily test non-UTC TZ
+    # Only test the impacted 43 files
+    # Split multiple parts to test to avoid OOM: refer to: https://github.com/NVIDIA/spark-rapids/issues/9829
```

Comment:
How long does this cost after adding new test suites?

cc @pxLi

Reply:
hmm seems adding too much duration. The ci_2 now cost ~3h5m even without default run_pyspark_from_build. for testing please also comment out cases after TZ ones to save time

Reply:
Here is testing the 43 files impacted by non-UTC time zone.
I'll select a sub-set of them.

Reply:
Per discussed offline, let's only include 1 test for each type of result. Minimal list:
- parquet scan
- orc scan
- csv scan
- cast
- one for TimeZoneAwareExpression
- one for UTCTimestamp

Reply:
super huge log which cannot be loaded in browser, I downloaded and checked TZ part

but we got some numbers for these TZ cases
```
start: [2023-11-24T01:47:31.478Z]
end: [2023-11-24T03:30:27.328Z]
```
~1h45m